modules/configuration/src/test/java/org/apache/ignite/internal/configuration/testframework/ConfigurationExtension.java

```diff
@@ -227,7 +228,10 @@ private static Object cfgValue(
         ConfigurationUtil.addDefaults(superRoot);
 
         if (!annotation.name().isEmpty()) {
-            superRoot.getRoot(rootKey).setInjectedNameFieldValue(annotation.name());
+            InnerNode root = superRoot.getRoot(rootKey);
+
+            root.internalId(UUID.randomUUID());
```

Comment:
Please add a test here `org.apache.ignite.internal.configuration.testframework.ConfigurationExtensionTest` to check if the behavior is correct.